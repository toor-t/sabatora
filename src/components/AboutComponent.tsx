/**
 * AboutComponent
 */

import * as React from 'react';
import Typography from '@material-ui/core/Typography';
import { withStyles, WithStyles, createStyles, Theme } from '@material-ui/core/styles';

import { Title, Str } from '../strings';
const img = require('../../build/icon.png');

const styles = () =>
    createStyles({
        div: {
            textAlign: 'center'
        },
        img: {
            width: '30%'
        }
    });

/**
 *  IAboutComponentStateProps
 */
export interface IAboutComponentStateProps {
    // TODO:
    // checked?: boolean;
}
/**
 * IAboutComponentDispatchProps
 */
export interface IAboutComponentDispatchProps {
    // onChange?: (e: any) => any;
}
/**
 * AboutComponent
 * @param props
 */
const AboutComponent: React.SFC<
    IAboutComponentStateProps & IAboutComponentDispatchProps & WithStyles<typeof styles>
> = props => {
    // TODO:
    return (
        <div>
            <div className={props.classes.div}>
                <img src={img} className={props.classes.img} />
            </div>
            <Typography
                variant="h6"
                color="inherit"
                style={{ flexGrow: 1 }}
                noWrap={true}
                align="center"
            >
                <br />
                {Str.AppName}
                <br />
                {Str.Copyright}
            </Typography>
        </div>
    );
};

export default withStyles(styles)(AboutComponent);
