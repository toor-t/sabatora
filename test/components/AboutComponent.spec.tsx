import * as React from 'react';
import * as renderer from 'react-test-renderer';

import AboutComponent from '../../src/components/AboutComponent';

describe('AboutComponent', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<AboutComponent />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
