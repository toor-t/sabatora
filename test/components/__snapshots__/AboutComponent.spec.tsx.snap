// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutComponent renders correctly 1`] = `
<div>
  <div
    className="AboutComponent-div-1"
  >
    <img
      className="AboutComponent-img-2"
      src="test-file-stub"
    />
  </div>
  <h6
    className="MuiTypography-root-3 MuiTypography-h6-20 MuiTypography-colorInherit-32 MuiTypography-noWrap-29 MuiTypography-alignCenter-26"
    style={
      Object {
        "flexGrow": 1,
      }
    }
  >
    <br />
    sabatora.
    <br />
    (C) 2008 toor-t
  </h6>
</div>
`;
